src/lib/Popup.svelte

```diff
@@ -179,6 +184,15 @@
     };
   });
 
+  function tryToOpen() {
+    if (canOpen && !canOpen(features)) {
+      open = false;
```

Comment:
A shared method like this is a nice approach. I had some bugs in a first attempt where I check and return early before setting `open = true`. But then there are bugs if you hover over something with a valid popup, then slide over to an adjacent feature without one -- the old popup stays open.